### PR TITLE
Osama bazbooz- APIs - Week2 - Homework

### PR DESCRIPTION
**Pull request by obazbooz**

### CODE REVIEW

**Review by jontra000 — CHANGES_REQUESTED** [review]
Hi @obazbooz. Exercises 1, 4, 5 and 6 all look good. There are just a few things to fix in exercises 2 and 3.

**Comment by jontra000** [comment]
I also noticed that when you press the 'Get Pokemon' button more than once the selector gets filled with multiple copies of the list. Could you fix this too please?

**Comment by obazbooz** [comment]
Dear @jontra000,

Thanks a lot for your valuable feedback on my homework.
I have fixed all the bugs you mentioned, it will be nice if you give me your feedback about the final result.

Best regards,
Osama

File: 3-UsingAPIs/Week2/homework/ex3-rollAnAce.js [hunk 1]
```diff
@@ -12,25 +12,23 @@ Full description at: https://github.com/HackYourFuture/Homework/blob/main/3-Usin
...
 const rollDice = require('../../helpers/pokerDiceRoller');
 
-async function rollDiceUntil(wantedValue, flag) {
+async function rollDiceUntil(wantedValue) {
   let rollDiceResponse = await rollDice();
   if (rollDiceResponse === wantedValue) {
     return rollDiceResponse;
   } else {
-    while (!flag) {
+    while (rollDiceResponse !== wantedValue) {
```
Comment (jontra000) [inline 3-UsingAPIs/Week2/homework/ex3-rollAnAce.js:20]: Notice how now the conditions in the if statements are just the negative of what is in the while statement, this means that the if statements are redundant and we can remove them and make this code a lot simpler.

**Review by jontra000 — CHANGES_REQUESTED** [review]
Good work fixing the bug. Could you address my two comments and then I can close the PR. Do let me know if you need any help

**Comment by obazbooz** [comment]
Dear @jontra000 ,

Thanks for your feedback, I have worked on the notes and fixed the fetch to be on click in ex2. Also I correct the if conditions statements bug in ex3. 
So hopefully it works well now  :D 
I will be happy to know your feedback about my fixing.

Best regards,
Osama

**Comment by jontra000** [comment]
Nice work @obazbooz. Everything looks good now

**Pull request closed without merging**